src/anm/events.js

```diff
@@ -164,28 +164,31 @@ MouseEventsSupport.prototype.dispatch = function(event) {
         return true;
     }
 
-    var dispatchedChild;  // found the matching child inside
-    if (owner.inside(localEvent)) {
-        owner.reverseEach(function(child) {
-            if (child.isActive()) {
-                dispatchedChild = child.dispatchMouseEvent(localEvent) ? child : null;
-                if (dispatchedChild) return false; // stop iteration of reverseEach
-            }
-        });
+    var dispatchedByOwner, // handled event myself
+        dispatchedByChild; // not handled myself, but found the matching child inside
 
-        if (!dispatchedChild) {
-            if (localEvent.type === 'mousemove') {
-                this.processMove(localEvent); // fire mouseenter/mouseexit if required
-            }
-            if ((localEvent.type === 'mouseclick') || (localEvent.type === 'mousedown')) {
-                state.pressedNode = this.owner;
-            }
-            this.owner.fire(localEvent.type, localEvent);
+    // here and below localEvent has properties `.x` and `.y`, so duck typing works
+    dispatchedByOwner = owner.inBounds && owner.inBounds(localEvent) && owner.inside && owner.inside(localEvent);
```

Comment:
matrices, curves are not relevant to mouse event processing at all

the spec contains the “transfromToChild” and “contains” method that cover
it all
it’s a spec, not an actual implementation
if I’d covered all of that I would have implemented it myself

On Tue, Jan 26, 2016 at 12:59 PM, Ulric Wilfred <notifications@github.com>
wrote:

> In src/anm/events.js
> <https://github.com/Animatron/player/pull/383#discussion_r50873203>:
>
> >
> > -        if (!dispatchedChild) {
> > -            if (localEvent.type === 'mousemove') {
> > -                this.processMove(localEvent); // fire mouseenter/mouseexit if required
> > -            }
> > -            if ((localEvent.type === 'mouseclick') || (localEvent.type === 'mousedown')) {
> > -                state.pressedNode = this.owner;
> > -            }
> > -            this.owner.fire(localEvent.type, localEvent);
> > +    // here and below localEvent has properties `.x` and `.y`, so duck typing works
> > +    dispatchedByOwner = owner.inBounds && owner.inBounds(localEvent) && owner.inside && owner.inside(localEvent);
>
> @kirillk <https://github.com/kirillk> there's no matrices in spec, no
> curves paths in spec, it's just an ideal-case example, I already moved
> everything to the code what was in this spec.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Animatron/player/pull/383/files#r50873203>.
>